### ISSUE TITLE
`kraken`: balance <-> position disparities with crypto/crypto pairs

### ISSUE THREAD

**Issue by goodboy** [pikers/piker#373]
Discovered by @iamzoltan and not surprising since i never tested any of this in the development of #349. 
The plan is to land that PR despite this issue (since master has a half working / bug filled order mode for `kraken` right now) and resolve this in post mortem fixes. 

----
#### issue descr
More or less there seems to be the [`ValueError` raised from the pp validation logic section of `trades_dialogue()`](https://github.com/pikers/piker/pull/349/files#diff-fdfbab77e4bbf9c1b5d79c4daa8dd93a498c497b35899bdc9bf5dc4dfadf587bR456) but reproducing this requires a ledger with the appropriate intercrypto pair transactions.

---
#### Solutions/TODO:
- [ ] one approach could be to have @iamzoltan submit an example ledger file / trades log + balance values that exemplifies the issue
  - hand over an example `trades_kraken_spot.toml` file
  - hand over the balance values from `Client.get_balances()` which can be hard coded manually in the code for testing
- [ ] @iamzoltan actually solves this directy 😂 

----
#### traceback
- [ ] this would be handy to have to get idea of the issue.

**Comment by goodboy** [comment]
I got something similar due to a couple different issues, so here is a (partial) follow up TODO list:

---
- [ ] for wallet transfers out of an account we get a `Transaction`-like entry in our `pps.toml` of the form:
```toml
[kraken.spot."xrpeur.kraken"]
size = 78.31125778
ppu = 0.3222278400000044
bsuid = "xrpeur"
clears = [
 { dt = "2022-12-21T07:20:16.316061+00:00", ppu = 0.3222278400000044, accum_size = 778.33125778, price = 0.3212, size = 778.33125778, cost = 0.4, tid = "TJKLRL-7UE2L-BX2UE5" },
 { dt = "2022-12-21T07:26:56+00:00", ppu = 0.3222278400000044, accum_size = 78.31125778, price = "NaN", size = -700.02, cost = 0, tid = "1503A92E24BAA1E8B8E7B78E79A78AAF3E6C3F3E35227EF7D2C13754876B37D5" },
]
```
  - **Notice the `price`** field has a `Nan` value which seemingly can break `.pp.py` clearing table calcs, though adjusting the clear iteration loop to sort on datetime seems to have resolved the crash?
  - [ ] we should probably put in a `type` field which can be filled with things like `'withdrawal'` or `'split'` or `'reverse_split'`, which i'm pretty sure is a bullet somewhere in https://github.com/pikers/piker/issues/345

---
- [ ] we aren't tabulating nor using the [widthdrawal fee schedule](https://docs.kraken.com/rest/#tag/User-Funding/operation/getWithdrawalInformation) in our `cost` field in each ledger entry, obviously this should resolve any accounting discrepancies a *transfer-as-sell* clear should have a cost (that kraken takes) as part of the position calc.

**Comment by goodboy** [comment]
ergg,

found another after a recent xrp transfer..

```python

Traceback (most recent call last):
  File "/home/goodboy/repos/tractor/tractor/_runtime.py", line 196, in _invoke
    res = await coro
  File "/home/goodboy/repos/piker/piker/brokers/kraken/broker.py", line 609, in trades_dialogue
    raise ValueError(
ValueError: Could not reproduce balance:
dst: xrp, 7.78e-06
```

**Comment by goodboy** [comment]
Closed in favor of #515
